craft_providers/bases/almalinux.py

```diff
@@ -95,7 +97,7 @@ def __init__(
 
         self._set_hostname(hostname)
 
-        self._packages: List[str] = []
+        self._packages: Optional[List[str]] = []
```

Comment:
This one will allow None?

Reply:
Yeah.  It should be safe to type it like this based on how is used.

One alternative is to change it in the parent class to `self._packages: List[str] = []`.  

It's an internal variable so either is fine with me if you have a preference.

Reply:
I think it is fine if the linters found we don't use it without check is None